Lib/platform.py

```diff
@@ -365,7 +365,10 @@ def win32_ver(release='', version='', csd='', ptype=''):
         return release, version, csd, ptype
 
     winver = getwindowsversion()
-    maj, min, build = winver.platform_version or winver[:3]
+    cmdwinver = _syscmd_ver()[2].split('.')
+    for x in range(3):
+        cmdwinver[x] = int(cmdwinver[x])
+    maj, min, build = tuple(cmdwinver) or winver[:3]
```

Comment:
```suggestion
    try:
        maj, min, build = tuple(map(int, _syscmd_ver()[2].split('.')))
    except ValueError:
        maj, min, build = winver.platform_version or winver[:3]
```

`_syscmd_ver()` doesn't validate that the version string has at least 3 dot-delimited numbers. Also, if it fails, it returns an empty string. In the above suggested code, both cases will raise `ValueError`, either from calling `int` on an empty string, or from trying to unpack too many values. It's handled by falling back on the old code that uses `platform_version`. This situation could occur due to problems with creating the shell process, or if the shell's output format changes in a new release and no longer matches the regex.

Reply:
Ok...let me commit your suggestion.